### DISCUSSION TITLE
Cache does not persist when used in macro's

### DISCUSSION THREAD

**Discussion by philipboeken** [spatie/once#65] [Bugs]
The cache does not persist within a single request when using once() in a macro function on the Laravel Eloquent Builder. For instance, if I register the following function in my Laravel application:
```
Illuminate\Database\Eloquent\Builder::macro('getOnce', function ($columns = ['*']) {
    return once(fn () => $this->get($columns));
});
```
then it would be nice if the cache persist when repeatedly calling `User::getOnce()` for example. Currently this is not the case, and I think this is caused by the repeated binding of the macro callback to the Builder object ([here](https://github.com/laravel/framework/blob/855a919d08b45f93cb3cf709736528c3d1531884/src/Illuminate/Database/Eloquent/Builder.php#L1639) or [here](https://github.com/laravel/framework/blob/855a919d08b45f93cb3cf709736528c3d1531884/src/Illuminate/Database/Eloquent/Builder.php#L1686)). I'm not too sure however how this works exactly. Could this possibly be mitigated by retrieving the Cache via Laravel's service provider?

**Comment by freekmurze** [comment]
I currently do not have the bandwidth to look into this, feel free to submit a PR with tests that fixes this.
